### PR TITLE
test: Test case for extended `SpriteButtonComponent`

### PR DESCRIPTION
**Pull request by puneet-739**
# Description

Added an extra test case for testing `SpriteButtonComponent` class. 
which includes a `CustomSpriteButtonComponet` class to test the working with `onLoad` instead of directly using Constructor.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

It may help working on the issues like  #3293 

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by spydon — COMMENTED** [review]
Tests for the `SpriteButtonComponent` should live in the `sprite_button_component_test.dart` file.

**Review by spydon — COMMENTED** [review]
These tests are just copies of already existing tests, but going through an extended version that isn't exposed to the user. I don't think we want this type of testing pattern in the code-base.
If you disagree with a good argument I can open it again, but I'll close it for now.

**Comment by puneet-739** [comment]
> These tests are just copies of already existing tests, but going through an extended version that isn't exposed to the user. I don't think we want this type of testing pattern in the code-base. If you disagree with a good argument I can open it again, but I'll close it for now.

Oh, ok.
Actually, I've added the test cases due to the issue #3293. (I think there is still an issue) Please refer to this comment https://github.com/flame-engine/flame/pull/3302#discussion_r1769626058
The reason of adding the test cases is to test the working of `SpriteButtonComponent` with extended version. because the issue #3293 only occurs when the user use the extended version but not on the Constructor.
So, we need to test if the issue is resolved for the extended version as well, as the present test cases only test for Constructor version, it will always give positive result (it gives error if we use the extended version).

**Comment by spydon** [comment]
@puneet-739 you're right, can you try https://github.com/flame-engine/flame/pull/3327? It should solve the issue so that the user doesn't have to call `super.onLoad` last.

**Comment by spydon** [comment]
@puneet-739 I changed my mind, due to the intricacies of the `SpriteButtonComponent` it's probably good to have these tests so that we don't get any regressions later. Can you fix the DCM warning? (Make the `CustomSpriteButtonComponent` private)

File: packages/flame/test/components/input/sprite_button_component_test.dart [hunk 1]
```diff
@@ -228,6 +404,69 @@ Future<void> main() async {
...
+      },
+    );
+  });
+}
+
+/// test class [_CustomSpriteButtonComponent] extending
+/// the [SpriteButtonComponent] class,
+/// To test [SpriteButtonComponent] without using Constructor
+/// but setting properties in [onLoad]
+///
```
Comment (spydon) [inline packages/flame/test/components/input/sprite_button_component_test.dart:447]:
```suggestion
/// This is used to test [SpriteButtonComponent] without using the constructor
/// and setting properties in [onLoad] of an extending class instead.
```

**Review by spydon — APPROVED** [review]

**Pull request merged**